src/main/client/src/components/@pages/ProfilePage.tsx

```diff
@@ -1,33 +1,104 @@
+import { ProfileDTO, RoleType } from 'types/dto/profile'
+import axios from 'axios'
+import { API_BASE_URL } from 'utils/configurations'
 
-const challenges = [
+const challenges = (profile: ProfileDTO) => [
   {
     name: 'Bucketlist',
-    value: 40
+    value: profile?.submittedAchievementCount,
+    total: profile?.totalAchievementCount
   },
   {
     name: 'Riddle',
-    value: 73
+    value: profile?.completedRiddleCount,
+    total: profile?.totalRiddleCount
   },
   {
     name: 'QR kód',
-    value: 17
+    value: profile?.tokens?.length,
+    total: profile?.minTokenToComplete
   }
 ]
 
 type ProfilePageProps = {}
 
+//Had to create a seperate skeleton layout so it wouldn't look strange
+export const ProfilePageSkeleton: React.FC<ProfilePageProps> = (props) => {
+  return (
+    <Page {...props}>
+      <VStack align="flex-start" mb="14" mt={6}>
+        <Skeleton h={12} w={['40%', null, null, '15%']}></Skeleton>
+        <Skeleton h={10} w={['50%', null, null, '20%']}></Skeleton>
+        <Skeleton h={10} w={['60%', null, null, '25%']}></Skeleton>
+      </VStack>
+      <Wrap spacing="3rem" justify="center" mt={3}>
+        {[0, 1, 2].map((challenge) => (
+          <WrapItem key={challenge}>
+            <Center w="10rem" h="12rem">
+              <Flex direction="column" align="center">
+                <Skeleton mb={3} w="90%" h={10} />
+                <SkeletonCircle size="10rem" />
+              </Flex>
+            </Center>
+          </WrapItem>
+        ))}
+      </Wrap>
+    </Page>
+  )
+}
+
 export const ProfilePage: React.FC<ProfilePageProps> = (props) => {
+  const [profile, setProfile] = React.useState<ProfileDTO | undefined>(undefined)
+  const [delay, setDelay] = React.useState<boolean>(false)
+
+  React.useEffect(() => {
+    //delay is needed to remove flickering of the skeleton
+    setDelay(true)
+    setTimeout(() => setDelay(false), 1000)
+    axios.get<ProfileDTO>(`${API_BASE_URL}/api/profile`).then((res) => {
+      if (res.status !== 200) {
+        console.log(res)
+        return
+      }
+      setProfile(res.data)
+    })
+  }, [setProfile])
+
+  if (delay || profile?.fullName === undefined) return <ProfilePageSkeleton {...props} />
+
   return (
     <Page {...props}>
       <Heading>Profil</Heading>
-      <Text fontSize="3xl">Kovács Péter</Text>
-      <Text marginBottom="10" fontSize="3xl">
-        Tankör: I06
-      </Text>
-      <Wrap spacing="3rem" justify="center">
-        {challenges.map((challenge) => (
+      <Text fontSize="3xl">{profile?.fullName}</Text>
+      <HStack justify="space-between">
+        <Text fontSize="3xl">Tankör: {profile?.groupName || 'nincs'}</Text>
+        {profile?.role && RoleType[profile.role] >= RoleType.STAFF && (
+          <a href="/admin/control/">
+            <Button as="span" colorScheme="red">
+              Admin panel
+            </Button>
+          </a>
+        )}
+      </HStack>
+      <Wrap spacing="3rem" justify="center" mt="10">
```

Comment:
This does not wrap for me. Please use the original layout as you can, because the mobile layout was great at that moment. Now it is not responsive for me.

Reply:
Seems to apear correctly on mobile, but when I try to resize it on desktop, it just overflows.

Reply:
I cannot reproduce the issue
![image](https://user-images.githubusercontent.com/23620633/150134572-5fc945af-d9d0-4cf4-9814-48ffbcf13b13.png)
![image](https://user-images.githubusercontent.com/23620633/150134615-fec2ebb7-080b-412f-bf79-aeed7a20bcd7.png)

Reply:
It might be just my browser, you proved it!